clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubBasedAdminClientManager.java

```diff
@@ -0,0 +1,296 @@
+            return Future.succeededFuture(subscription.getSubscription());
+        }
+        try {
+            final Subscription request = Subscription.newBuilder()
+                    .setName(subscription.toString())
+                    .setTopic(topic.toString())
+                    .setPushConfig(PushConfig.getDefaultInstance())
+                    .setAckDeadlineSeconds(0)
+                    .setMessageRetentionDuration(Durations.fromMillis(MESSAGE_RETENTION))
+                    .build();
+            final Subscription createdSubscription = subscriptionAdminClient.createSubscription(request);
+            if (createdSubscription == null) {
+                LOG.error("Creating subscription failed [subscription: {}, topic: {}, project: {}]", subscription,
+                        topic,
+                        projectId);
+                return Future.failedFuture("Subscription creation failed.");
+            }
+            subscriptions.add(createdSubscription.getName());
+            return Future.succeededFuture(subscription.getSubscription());
+        } catch (ApiException e) {
+            LOG.error("Error creating subscription {} for topic {} on project {}", subscription, topic,
+                    projectId, e);
+            return Future.failedFuture("Subscription creation failed.");
+        }
+    }
+
+    private boolean subscriptionExists(final String subscriptionName) {
+        if (subscriptions.contains(subscriptionName)) {
+            return true;
+        }
+        final ProjectName projectName = ProjectName.of(projectId);
+        try {
+            final ListSubscriptionsPagedResponse pagedResponse = subscriptionAdminClient.listSubscriptions(projectName);
+            if (pagedResponse != null) {
+                for (Subscription subscription : pagedResponse.iterateAll()) {
+                    if (subscription.getName().equals(subscriptionName)) {
+                        subscriptions.add(subscriptionName);
+                        return true;
+                    }
+                }
+            }
+            return false;
+        } catch (Exception e) {
+            LOG.error("Error listing subscriptions on project {}", projectId, e);
+            return false;
+        }
+    }
+
+    private void closeSubscriptionAdminClient() {
+        if (subscriptionAdminClient != null) {
+            subscriptionAdminClient.shutdown();
+            try {
+                subscriptionAdminClient.awaitTermination(1, TimeUnit.MINUTES);
+            } catch (InterruptedException e) {
+                LOG.debug("Resources are not freed properly, error", e);
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    private void closeTopicAdminClient() {
+        if (topicAdminClient != null) {
+            topicAdminClient.shutdown();
+            try {
+                topicAdminClient.awaitTermination(1, TimeUnit.MINUTES);
```

Comment:
what purpose does it serve to wait for a minute here?

Reply:
updated it to 5 seconds (same as _PubSubPublisherClient_)